probe_scraper/remote_storage/gcs.py

```diff
@@ -0,0 +1,64 @@
+from pathlib import Path, WindowsPath
+from typing import Optional
+
+from google.cloud import storage
+
+from .utils import compile_glob, extract_bucket_and_prefix, path_walk
+
+
+def pull(src: str, dst: Path):
+    bucket_name, prefix = extract_bucket_and_prefix(src)
+    prefix_length = len(prefix)
+    bucket = storage.Client().get_bucket(bucket_name)
+    # TODO verify that this correctly handles prefix being a file
+    for blob in bucket.list_blobs(prefix=prefix):
```

Comment:
Would it make sense to use `gsutil rsync` to simplify this, similar to `aws s3 sync` in the other backend?

Reply:
Unfortunately `gsutil` doesn't expose flags for setting content-encoding and content-type, but the sdk does.

Reply:
turns out it does support those things, via `-h`. I also thought that gsutil wasn't a python package, but it turns out I was also wrong about that.

I'm working on changing this to use `gsutil`